tests/integration_test/service_control_failed_request_report_test.go

```diff
@@ -85,6 +85,68 @@ func TestServiceControlFailedRequestReport(t *testing.T) {
 				},
 			},
 		},
+		{
+			desc:           "Request matches uri template(exact path) but not method.",
+			url:            fmt.Sprintf("localhost:%v", s.Ports().ListenerPort),
+			clientProtocol: "http",
+			// "DELETE" is not defined for "/v1/shelves".
+			httpMethod:    "DELETE",
+			method:        "/v1/shelves?key=api-key",
+			httpCallError: "405 Method Not Allowed, {\"code\":405,\"message\":\"The current request is matched to the defined url template \"/v1/shelves\" but its http method is not allowed\"}",
+			wantScRequests: []interface{}{
+				&utils.ExpectedReport{
+					Version:         utils.ESPv2Version(),
+					ServiceName:     "bookstore.endpoints.cloudesf-testing.cloud.goog",
+					ServiceConfigID: "test-config-id",
+					URL:             "/v1/shelves?key=api-key",
+					ApiMethod:       "<Unknown Operation Name>",
+					// API Key is extracted but not trusted.
+					ApiKeyInLogEntryOnly: "api-key",
+					ApiKeyState:          "NOT CHECKED",
+					ProducerProjectID:    "producer project",
+					FrontendProtocol:     "http",
+					HttpMethod:           "DELETE",
+					LogMessage:           "<Unknown Operation Name> is called",
+					StatusCode:           "0",
+					ResponseCode:         405,
+					Platform:             util.GCE,
+					Location:             "test-zone",
+					BackendProtocol:      "grpc",
+					ResponseCodeDetail:   "direct_response",
+				},
+			},
+		},
+		{
+			desc:           "Request matches uri template(regex) but not method.",
+			url:            fmt.Sprintf("localhost:%v", s.Ports().ListenerPort),
+			clientProtocol: "http",
+			// "DELETE" is not defined for "/v1/shelves".
+			httpMethod:    "POST",
+			method:        "/v1/shelves/100?key=api-key",
```

Comment:
could we add two more cases for trailing_slash, one is exact match and another regex?

Reply:
Done.